File: app/src/main/java/sgtmelon/scriptum/app/screen/note/NoteCallback.java
package sgtmelon.scriptum.app.screen.note;

import androidx.annotation.NonNull;
import sgtmelon.scriptum.app.control.SaveControl;

/**
 * Интерфейс общения с {@link NoteActivity}
 */
public interface NoteCallback {

    void setupFragment(boolean isSave);

    @NonNull
    SaveControl getSaveControl();

    @NonNull
    NoteViewModel getViewModel();

}

File: app/src/main/java/sgtmelon/scriptum/app/screen/note/NoteCallback.kt
package sgtmelon.scriptum.app.screen.note

/**
 * Интерфейс общения [NoteViewModel] с [NoteActivity]
 */
interface NoteCallback {

    fun showTextFragment(id: Long, isSave: Boolean)

    fun showRollFragment(id: Long, isSave: Boolean)

    fun onPressBackText(): Boolean

    fun onPressBackRoll(): Boolean

    fun finish()

}